django/forms/boundfield.py

```diff
@@ -219,12 +219,28 @@ def initial(self):
         return data
 
     def build_widget_attrs(self, attrs, widget=None):
-        widget = widget or self.field.widget
         attrs = dict(attrs)  # Copy attrs to avoid modifying the argument.
-        if widget.use_required_attribute(self.initial) and self.field.required and self.form.use_required_attribute:
-            attrs['required'] = True
+        if isinstance(widget, MultiWidget) or isinstance(self.field.widget, MultiWidget):
+            widgets = widget.widgets or self.field.widget.widgets
+            for field, widget in zip(self.field.fields, widgets):
+                if (
+                    widget.use_required_attribute(self.initial) and
+                    field.required and
+                    self.form.use_required_attribute
+                ):
+                    widget.attrs["required"] = True
+                else:
+                    widget.attrs["required"] = False
+        else:
+            widget = widget or self.field.widget
+            if (
+                widget.use_required_attribute(self.initial) and
+                self.field.required and
+                self.form.use_required_attribute
+            ):
+                attrs["required"] = True
```

Comment:
We need to factor this differently, as it's an exact duplicate of the added block above.